### PR TITLE
Recognize constants with alias type

### LINKED ISSUE #101: static const with in-class initializer should be converted to a constant rather than extern static
Currently, static consts in class are all converted to `extern "C" pub static`, however, when the static const has in-class initializer, we should be able to convert it to a constant like macro.

FWIW, C++ only allows static const which has integer type or enum type to have in-class initializer.

### PR DESCRIPTION
**Pull request by upsuper**
This fixes #101.

### CODE REVIEW

File: src/ir/ty.rs [hunk 1]
```diff
@@ -104,9 +104,13 @@ impl Type {
         Self::new(Some(name), None, kind, false)
     }
 
-    pub fn is_integer_literal(&self) -> bool {
+    pub fn is_integer(&self, ctx: &BindgenContext) -> bool {
         match *self.kind() {
             TypeKind::Int(..) => true,
+            TypeKind::ResolvedTypeRef(inner) |
+            TypeKind::Alias(_, inner) => {
```
Comment (emilio) [inline src/ir/ty.rs]: Instead of modifying this, can you use `t.canonical_type(ctx).is_integer_literal()` instead?

Also, could you add a test for this?

Comment (upsuper) [inline src/ir/ty.rs]: Yeah, using `canonical_type` should be better.

The issue for adding a test is that... the original test fails even on master for me... :(

Comment (emilio) [inline src/ir/ty.rs]: Yeah, if you have llvm 3.9 `llvm_stable` will fail, that's known. If you take care of uploading the test I can update the expectations before merging myself when I arrive home.

Comment (emilio) [inline src/ir/ty.rs]: Yeah, on Mac you need to `brew install llvm38` at least (and then maybe configure paths and all, I don't remember correctly)

**Comment by upsuper** [comment]
The third commit was changed significantly because of issues revealed by my test.

**Comment by upsuper** [comment]
Yeah, I'm looking into the failures...

**Comment by upsuper** [comment]
Unfortunately, even with llvm38, the test doesn't run as expected on my mac... It still crashes. So I have to push to see whether there is still any test failure :(

**Comment by upsuper** [comment]
Looks like all green now \o/

**Comment by upsuper** [comment]
But you may want to re-review the second and third commit.

**Comment by upsuper** [comment]
Ah, the first and the third.

File: src/ir/ty.rs [hunk 2]
```diff
@@ -104,10 +104,13 @@ impl Type {
         Self::new(Some(name), None, kind, false)
     }
 
-    pub fn is_integer_literal(&self) -> bool {
-        match *self.kind() {
-            TypeKind::Int(..) => true,
-            _ => false,
+    pub fn is_integer(&self, ctx: &BindgenContext) -> bool {
+        match self.kind {
+            TypeKind::UnresolvedTypeRef(..) => false,
```
Comment (emilio) [inline src/ir/ty.rs]: Heh, I was surprised we didn't run into this.

File: src/ir/var.rs [hunk 3]
```diff
@@ -169,3 +165,22 @@ fn parse_int_literal_tokens(cursor: &clang::Cursor,
...
+fn get_integer_literal_from_cursor(cursor: &clang::Cursor,
+                                   unit: &clang::TranslationUnit) -> Option<i64> {
+    use clangll::*;
+    let mut value = None;
+    cursor.visit(|c, _| {
+        match c.kind() {
+            CXCursor_IntegerLiteral => {
+                value = parse_int_literal_tokens(&c, unit);
+            }
+            CXCursor_FirstExpr => {
```
Comment (emilio) [inline src/ir/var.rs]: Can you use `UnexposedExpr` instead? First/Last is just the way clang gives you to iterate over them.

File: src/ir/var.rs [hunk 4]
```diff
@@ -169,3 +165,22 @@ fn parse_int_literal_tokens(cursor: &clang::Cursor,
...
+                value = get_integer_literal_from_cursor(&c, unit);
+            }
+            _ => ()
+        }
+        CXChildVisit_Continue
```
Comment (emilio) [inline src/ir/var.rs]: Also, probably you want something here like `if value.is_some() { CXChildVisit_Break } else { CXChildVisit_Continue }`.

File: src/ir/var.rs [hunk 5]
```diff
@@ -141,31 +137,60 @@ fn parse_int_literal_tokens(cursor: &clang::Cursor,
...
+                                   unit: &clang::TranslationUnit) -> Option<i64> {
+    use clangll::*;
+    let mut value = None;
+    cursor.visit(|c, _| {
+        match c.kind() {
+            CXCursor_IntegerLiteral |
+            CXCursor_UnaryOperator => {
+                value = parse_int_literal_tokens(&c, unit);
+            }
+            CXCursor_FirstExpr => {
```
Comment (emilio) [inline src/ir/var.rs]: Can you use `UnexposedExpr` instead? First/Last are meant just to iterate.

**Comment by emilio** [comment]
r=me with that nit, thanks for doing this!

@bors-servo: delegate+

**Pull request merged**